### PR TITLE
Fix InstructionTree and Block circular dependency

### PR DESCRIPTION
**Pull request by dbnicholson**
Since 4338632 there's been a circular dependency between InstructionTree and Block that manifests as a script error due to a non-existent resource in a seemingly unrelated object. Several components use InstructionTree.TreeNode including Block, and prior to 4338632 there were no references to any non-native classes. Split out the block tree functions to a new BlockTreeUtil class so that InstructionTree no longer contains references to Block.

### CODE REVIEW

**Comment by dbnicholson** [comment]
This will almost certainly cause conflicts in the decoupling PRs.

File: addons/block_code/drag_manager/drag.gd [hunk 1]
```diff
@@ -3,6 +3,7 @@ extends Control
 
 const Background = preload("res://addons/block_code/ui/blocks/utilities/background/background.gd")
 const BlockCanvas = preload("res://addons/block_code/ui/block_canvas/block_canvas.gd")
+const BlockTreeUtil = preload("res://addons/block_code/ui/block_tree_util.gd")
```
Comment (dbnicholson) [inline addons/block_code/drag_manager/drag.gd:6]: I'm not at all convinced of this name or path. What seemed to be in common in the functions moved was that they were about a tree of `Block` nodes. @wnbaum's work completely changes the way this is handled.

**Comment by manuq** [comment]
> This will almost certainly cause conflicts in the decoupling PRs.

No problem, we can rebase.

File: addons/block_code/drag_manager/drag.gd (hunk 1 above)
Comment (manuq) [reply addons/block_code/drag_manager/drag.gd:6]: If this is going to be changed anyways maybe the current name/path doesn't matter that much?

**Review by manuq — APPROVED** [review]
Excellent, this fixes the circular dependency in a decent way with a separate script. I tested it in both the development project and by copying the plugin to a new project and then enabling it. No errors present.

**Pull request merged**